cwf/cloud/go/services/cwf/servicers/builder_servicer.go

```diff
@@ -223,6 +217,25 @@ func getHealthServiceGrePeers(pipelinedPeers []*lte_mconfig.PipelineD_AllowedGre
 	return healthPeers
 }
 
+func getHAPairConfigs(graph configurator.EntityGraph, gatewayID string) *models.CwfHaPairConfigs {
+	haPairEnts := graph.GetEntitiesOfType(cwf.CwfHAPairType)
```

Comment:
There's `GetFirstAncestorOfType` in the `configurator` package, or maybe one of the other functions will be more useful here.